src/BaseItem.vala

```diff
@@ -90,6 +137,26 @@ public class Dock.BaseItem : Gtk.Box {
 
         gesture_click = new Gtk.GestureClick ();
         add_controller (gesture_click);
+
+        if (disallow_dnd) {
+            return;
+        }
+
+        var drag_source = new Gtk.DragSource () {
+            actions = MOVE
+        };
+        add_controller (drag_source);
+        drag_source.prepare.connect (on_drag_prepare);
+        drag_source.drag_begin.connect (on_drag_begin);
+        drag_source.drag_cancel.connect (on_drag_cancel);
+        drag_source.drag_end.connect (on_drag_end);
+
+        var drop_target = new Gtk.DropTarget (get_type (), MOVE) {
+            preload = true
+        };
+        add_controller (drop_target);
+        drop_target.enter.connect (on_drop_enter);
+        drop_target.drop.connect (on_drop);
```

Comment:
I think we still want to add drop target even when dran-and-drop isn't allowed

Reply:
Hmmm why?

Reply:
Because dragging a workspace to the left of dynamic workspace icon doesn't do anything, while doing so on icon groups makes space for dragged workspace

Reply:
Ah but just adding the drop target wont fix it since it only accepts drops from the same type

Reply:
Ok we support now a drag to the dynamic workspace icon.